typescript/infra/scripts/list-validator-checkpoint-indices.ts

```diff
@@ -0,0 +1,47 @@
+import { ChainNameToDomainId } from '@hyperlane-xyz/sdk';
+
+import { S3Validator } from '../src/agents/aws/validator';
+import { concurrentMap } from '../src/utils/utils';
+
+import {
+  getContextAgentConfig,
+  getCoreEnvironmentConfig,
+  getEnvironment,
+} from './utils';
+
+async function main() {
+  const environment = await getEnvironment();
+  const config = getCoreEnvironmentConfig(environment);
+
+  const agentConfig = await getContextAgentConfig(config, 'abacus');
+
+  const indices = await concurrentMap(
+    4,
+    Object.entries(agentConfig.validatorSets).flatMap(([chain, set]) =>
+      set.validators.map((validator) => ({ chain, validator })),
+    ),
+    async ({ chain, validator }) => {
+      const s3Validator = new S3Validator(
+        validator.address,
+        // @ts-ignore
+        ChainNameToDomainId[chain],
+        // @ts-ignore
+        `https://${validator.checkpointSyncer.bucket!}.s3.${
+          // @ts-ignore
+          validator.checkpointSyncer.region
```

Comment:
are we not doing this somewhere else already?

Reply:
Interestingly no. The existing validator script just takes in the URL as a parameter and the agent actually takes bucket and region separately